Dshop/apps/products_catalogue/views.py

```diff
@@ -47,3 +48,46 @@ def get(self, request, *args, **kwargs):
 
         xml_response = self.generate_xml_file_for_ceneo(products)
         return xml_response
+
+def GetCeneoCategories(request):
+    
+    # Pobierz XML
+    url = 'https://developers.ceneo.pl/api/v3/kategorie'
+    response = requests.get(url)
+    xml_data = response.content
+    
+    root  = ET.fromstring(xml_data)
+    
+
+    def parse_category(category_elem):
+        category = {
+            'Id': category_elem.find('Id').text,
+            'name': category_elem.find('Name').text,         
+        }
+        return category
+
+    def parse_categories(category_elem):
+        categories = []
+        for elem in category_elem:
+            category = parse_category(elem)
+            subcategories_elem = elem.find('Subcategories')
+            if subcategories_elem is not None:
+                categories.extend(parse_categories(subcategories_elem))
+            categories.append(category)
+        return categories
+    
+
+    categories = parse_categories(root)
+
+    for x in categories:
+        x['Id']=int(x['Id'])
+
+    categories.sort(key = lambda x: x['Id'])
+
+    for category in categories:
+        cc = CeneoCategory(id=category['Id'], name=category['name'])
+        cc = CeneoCategory(id=category['Id'], name=category['name'])
+        cc.save()
```

Comment:
Please use bulk create. This will be much easier on DB, 
https://docs.djangoproject.com/en/4.2/ref/models/querysets/#bulk-create

Reply:
done